### ISSUE TITLE
MD053 problem with link definition usage over two lines in blockquotes

### ISSUE THREAD

**Issue by ericcornelissen** [DavidAnson/markdownlint#544]
Rule [MD053](https://github.com/DavidAnson/markdownlint/blob/cba2ca0dbdc8f11e4b97c054380ffd52bb7aae19/doc/Rules.md#md053---link-and-image-reference-definitions-should-be-needed) incorrectly complains about an unused link definition when the link usage:

1. Spans 2 lines
2. Is in a blockquote

## Reproducing

The following MarkDown file should reproduce the error:

```markdown
<!-- error.md -->

# Example Blockquotes

Example 1: newline in link, working

> Find out more in the GitHub Actions documentation on [`push` events], [branch
> and tag filters], and [path filters].

Example 2: newline in link, not working

> Find out more in the GitHub Actions documentation on [`repository_dispatch`
> events] and [`workflow_dispatch` events].

[`pull_request` events]: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request
[`push` events]: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#push
[`repository_dispatch` events]: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#repository_dispatch
[`workflow_dispatch` events]: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#workflow_dispatch
[branch and tag filters]: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags
[path filters]: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths
```

Result:

```sh
$ npx markdownlint error.md
error.md:20:1 MD053/link-image-reference-definitions Link and image reference definitions should be needed [Unused link or image reference definition: "`repository_dispatch` events"] [Context: "[`repository_dispatch` events]..."]
```

<details>
<summary>When removing the blockquotes, the problem is no longer reporter.</summary>

```markdown
<!-- success.md -->

# Example No Blockquotes

Example 1: newline in link, working

Find out more in the GitHub Actions documentation on [`push` events], [branch
and tag filters], and [path filters].

Example 2: newline in link, not working

Find out more in the GitHub Actions documentation on [`repository_dispatch`
events] and [`workflow_dispatch` events].

[`pull_request` events]: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request
[`push` events]: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#push
[`repository_dispatch` events]: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#repository_dispatch
[`workflow_dispatch` events]: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#workflow_dispatch
[branch and tag filters]: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags
[path filters]: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths
```

Result:

```sh
$ npx markdownlint success.md
```

</details>

_(These examples are based on the documentation for [ericcornelissen/svgo-action](https://github.com/ericcornelissen/svgo-action), where I first encountered this bug.)_

**Comment by ericcornelissen** [comment]
My bad :sweat_smile: Thanks for double checking it! I updated the original issue description to avoid confusion.
